### PR TITLE
Overhaul ocm mapping

### PR DESCRIPTION
**Pull request by ccwienk**

### CODE REVIEW

File: concourse/steps/component_descriptor.mako [hunk 1]
```diff
@@ -357,3 +358,36 @@ for idx, component_id in enumerate(cnudie.purge.iter_componentversions_to_purge(
...
+% if not ocm_repository_mappings:
+  return
+% endif
+% for mapping in ocm_repository_mappings:
+  % if not mapping.prefix:
+  yield '${mapping.repository}'
+  % else:
+  component_name = cnudie.util.to_component_name(component)
+  if component_name.startswith('${mapping.prefix}'):
+   yield '${mapping.repository}'
```
Comment (zkdev) [inline concourse/steps/component_descriptor.mako:389]: not properly indented

Comment (ccwienk) [reply concourse/steps/component_descriptor.mako:389]: you mean rather use two blanks?
kk

File: cnudie/retrieve.py [hunk 2]
```diff
@@ -72,15 +68,6 @@ def _iter_ocm_repositories(
         yield repository_cfg
         return
 
-    if isinstance(repository_cfg, cnudie.util.OcmResolverConfig):
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply cnudie/retrieve.py:75]: good catch!
of course not

[the remark this answers is not in the document]
Comment (ccwienk) [reply cnudie/retrieve.py:75]: however, I think this refers to a piece of code that is effectively not the final state of this pull-request 🤔

File: concourse/model/traits/component_descriptor.py [hunk 3]
```diff
@@ -49,6 +47,12 @@ class UploadMode(enum.StrEnum):
     NO_UPLOAD = 'no-upload'
 
 
+@dataclasses.dataclass(frozen=True)
+class OcmRepositoryMappingEntry:
+    repository: str
+    prefix: str | None = None
```
[the remark this answers is not in the document]
Comment (ccwienk) [reply concourse/model/traits/component_descriptor.py:53]: not needed. it is never used AFAIK in our pipeline-definitions. we do not have a single use-case where ordering is not enough

**Pull request merged**